node/src/reactor/participating.rs

```diff
@@ -909,6 +911,13 @@ impl reactor::Reactor for Reactor {
                 let mut effects =
                     self.dispatch_event(effect_builder, rng, Event::DeployGossiper(event));
 
+                let event = event_stream_server::Event::DeployAccepted(*deploy.id());
```

Comment:
I read the description of the PR about not allocating too much memory but I am not sure I understand why. We should accept every deploy only once and immediately emit it to the event stream. Either there is a consumer subscribed – in which case we emit the deploy to the event and drop the deploy, or not – in which case we can drop the deploy immediately. 

Why do you think sending a whole deploy in `Event::DeployAccepted` could be a problem?

Reply:
I think I see it now – b/c we keep `n` last events in the ring buffer of the event server, it's possible that all of those slots become occupied with `DeployAccepted` events which, if they were wrapping whole `Deploy` instances, could start occupying a lot of memory.

Reply:
Yup - exactly.